user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserOperator.java

```diff
@@ -139,12 +149,111 @@ protected void createOrUpdate(Reconciliation reconciliation, KafkaUser kafkaUser
             scramAcls = user.getSimpleAclRules();
         }
 
+        KafkaUserStatus userStatus = new KafkaUserStatus();
         CompositeFuture.join(
                 scramShaCredentialOperator.reconcile(user.getName(), password),
-                secretOperations.reconcile(namespace, user.getSecretName(), desired),
+                reconcileSecretAndSetStatus(namespace, user, desired, userStatus),
                 aclOperations.reconcile(KafkaUserModel.getTlsUserName(userName), tlsAcls),
                 aclOperations.reconcile(KafkaUserModel.getScramUserName(userName), scramAcls))
-                .map((Void) null).setHandler(handler);
+                .setHandler(reconciliationResult -> {
+
+                    if (kafkaUser.getMetadata().getGeneration() != null)    {
+                        userStatus.setObservedGeneration(kafkaUser.getMetadata().getGeneration());
+                    }
+
+                    Condition readyCondition = ConditionUtils.buildConditionFromReconciliationResult(reconciliationResult.mapEmpty());
+
+                    userStatus.setUsername(user.getName());
+                    userStatus.setConditions(Collections.singletonList(readyCondition));
+
+                    updateStatus(kafkaUser, reconciliation, userStatus).setHandler(statusResult -> {
+                        if (statusResult.succeeded())    {
+                            log.debug("Status for {} is up to date", kafkaUser.getMetadata().getName());
```

Comment:
* Can we use the reconciliation here in the log message as we do with other logs?
* Is this message really needed? It seems to be a bit duplicate to the messages inside the `updateStatus` method.